torch/ao/quantization/fx/_model_report/README.md

```diff
@@ -0,0 +1,119 @@
+
+## Usage
+
+This snippet should be ready to copy, paste, and use with the exception of a few small parts denoted in `#TODO` comments
+
+```python
+# prep model
+q_config_mapping = torch.ao.quantization.get_default_qconfig_mapping() # alternatively use your own qconfig mapping if you alredy have one
+model = Model() # TODO define model
+example_input = model.get_example_data()[0] # get example data for your model
+prepared_model = quantize_fx.prepare_fx(model, q_config_mapping, example_input)
+
+# create ModelReport instance and insert observers
+detector_set = set([PerChannelDetector(), InputWeightDetector(0.5), DynamicStaticDetector(), OutlierDetector()]) # TODO add all desired detectors
+model_report = ModelReport(detector_set)
+ready_for_callibrate = model_report.prepare_detailed_callibration(prepared_model)
+
+# TODO run callibration of model with relavent data
+
+# generate reports for your model and remove observers if desired
+reports = model_report.generate_model_report(ready_for_callibrate, remove_inserted_observers=True)
+for report_name in report.keys():
+    text_report, report_dict = reports[report_name]
+    print(text_report, report_dict)
+
+# TODO update q_config_mapping based on feedback from reports
+```
+
+There is a tutorial in the works that will walk through a full usage of the ModelReport API.
+This tutorial will show the ModelReport API being used on toy model in both an Fx Graph Mode workflow and an alterative workflow with just a traceable model.
+This README will be updated with a link to the tutorial upon completion of the tutorial.
+
+## Implementing Your Own Detector
+
+The main way to add functionality to the ModelReport API is to add more Detectors.
+Detectors each have a specific focus in terms of the type of information they collect information on.
+For example, the `DynamicStaticDetector` figures out whether Dynamic or Static Quantization is appropriate for different layers.
+Meanwhile, the `InputWeightEqualizationDetector` determines whether Input-Weight Equalization should be applied for each layer.
+
+
+### Requirements to Implement Detector
+All Detectors inherit from the `DetectorBase` class, and all of them (including any custom detectors you create) will need to implement 3 methods:
+- `determine_observer_insert_points(self, model)` -> `Dict`: determines which observers you want into insert into a model to gather statistics and where in the model. All of them return a dictionary mapping unique observer fully qualified names (fqns), which is where we want to insert them, to a dictionary of location and argument information in the format:
+
+```python
+return_dict = {
+    "[unique_observer_fqn_of_insert_location]" :
+    {
+        "target_node" -> the node we are trying to observe with this observer (torch.fx.node.Node),
+        "insert_observer" -> the observer we wish to insert (ObserverBase),
```

Comment:
I feel `insert_observer` is not very intuitive, maybe rename this to `observer_to_insert` or something similar?

Reply:
Addressed in new downstream commit here: https://github.com/pytorch/pytorch/pull/81382